### PR TITLE
Out of date govmomi being used in libretto.

### PR DESCRIPTION
**Pull request by avnish30jn**
**Jira Id**
[C3-100](https://apporbit.atlassian.net/browse/C3-100)

**Problem**

Govmomi in libretto is out of date. Updating govmomi needs some code changes in libretto.

**Solution**

Changed the libretto code to work with govmomi, Context updated similar to govmomi, UnitNumber now returning reference, hence removing '&', Create disk takes one more parameter, datastore ManagedObjectReference, Conflicting int and int32

### CODE REVIEW

File: virtualmachine/vsphere/util.go [hunk 1]
```diff
@@ -16,7 +16,7 @@ import (
 	"regexp"
 	"strings"
```
Comment (pankajtandon12) [inline virtualmachine/vsphere/util.go:19]: remove this empty line

File: virtualmachine/vsphere/vm.go [hunk 2]
```diff
@@ -22,7 +22,7 @@ import (
 	"github.com/vmware/govmomi/property"
 	"github.com/vmware/govmomi/vim25/types"
 
-	"golang.org/x/net/context"
+	"context"
```
Comment (pankajtandon12) [inline virtualmachine/vsphere/vm.go]: move this up, along with rest of the standard library

File: virtualmachine/vsphere/vm.go [hunk 3]
```diff
@@ -50,7 +50,7 @@ type VMwareLease struct {
 // HTTPNfcLeaseProgress takes a percentage as an int and sets that percentage as
 // the completed percent.
 func (v VMwareLease) HTTPNfcLeaseProgress(p int) {
-	v.Lease.HttpNfcLeaseProgress(v.Ctx, p)
+	v.Lease.HttpNfcLeaseProgress(v.Ctx, int32(p))
```
Comment (pankajtandon12) [inline virtualmachine/vsphere/vm.go]: This change should be propagated to callers as well.
i.e.
func (v VMwareLease) HTTPNfcLeaseProgress(p int32) {

File: virtualmachine/vsphere/util.go [hunk 4]
```diff
@@ -419,13 +419,23 @@ var reconfigureVM = func(vm *VM, vmMo *mo.VirtualMachine) error {
 		return err
 	}
 
+	dcMo, err := GetDatacenter(vm)
+	if err != nil {
+		return fmt.Errorf("Failed to retrieve datacenter: %s", err)
```
Comment (pankajtandon12) [inline virtualmachine/vsphere/util.go]: use `return err`, same as rest of the code. Assuming caller is expecting/handling that.
You can still print the user friendly failure message before returning.

File: virtualmachine/vsphere/util.go [hunk 5]
```diff
@@ -419,13 +419,23 @@ var reconfigureVM = func(vm *VM, vmMo *mo.VirtualMachine) error {
...
+	}
+
+	dsMo, err := findDatastore(vm, dcMo, vm.Datastores[0])
```
Comment (pankajtandon12) [inline virtualmachine/vsphere/util.go]: Missing error handling for vm.Datastores[0].
Is it assumed to be handle in caller? 
Generic: For all param validations, if it is assumed to be handle by caller, Add a comment to inform that explicitly.

**Review by pankajtandon12 — CHANGES_REQUESTED** [review]
@avnish30jn Please find some review comments.
If not already, please ensure that corresponding changes are done for all instances of issue.

File: virtualmachine/vsphere/util.go (hunk 1 above)
Comment (avnish30jn) [reply virtualmachine/vsphere/util.go:19]: This is corrected

File: virtualmachine/vsphere/util.go (hunk 4 above)
Comment (avnish30jn) [reply virtualmachine/vsphere/util.go]: This is corrected

File: virtualmachine/vsphere/util.go (hunk 5 above)
Comment (avnish30jn) [reply virtualmachine/vsphere/util.go]: This is corrected

File: virtualmachine/vsphere/vm.go (hunk 2 above)
Comment (avnish30jn) [reply virtualmachine/vsphere/vm.go]: This is corrected

File: virtualmachine/vsphere/vm.go (hunk 3 above)
Comment (avnish30jn) [reply virtualmachine/vsphere/vm.go]: This is corrected

**Review by pankajtandon12 — APPROVED** [review]

**Pull request merged**